### PR TITLE
build multiarch CI containers, use podman and run user-level tests

### PR DESCRIPTION
**Pull request by jluebbe**
This PR refactors our CI container builds to use podman instead of docker.

It also adds pre-built containers for arm/v5, arm/v7, arm64/v8 and 386, which allows us to drop the dependency installation under QEMU.

For now, we need to install a newer buildah/podman from Ubuntu noble, but we'll be able to drop that step when GitHub Actions adds an image for the Ubuntu 24.04 LTS release.

To show how the containers work, I've added one commit to pull them from jluebbe/rauc instead of the main repo. When this PR is reviewed, I'd drop that commit.

### CODE REVIEW

File: test/Dockerfile [hunk 1]
```diff
@@ -62,10 +62,6 @@ RUN git clone https://github.com/folbricht/desync.git /tmp/desync && \
 RUN mkdir -p /lib/modules && \
     mkdir -p /var/run/dbus
 
-# We want to run as non-root user equaling uid of Travis' user 'travis' (2000)
-ENV user travis
+RUN adduser --disabled-password --gecos "" rauc-ci
```
Comment (ejoerns) [inline test/Dockerfile]: `man adduser` doesn't list `--gecos` anymore, but for `--comment` it says:

```man
       --comment comment
              Set  the  comment  field for the new entry generated.  adduser will not ask for the information if this option is given.  This field is also known under the name GECOS field and contains information that is used by the finger(1) com-
              mand.  This used to be the --gecos option, which is deprecated and will be removed after Debian bookworm.  Valid Modes: adduser, adduser --system.
```

File: test/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,9 @@
 FROM debian:bullseye
 
 # Required for building
-RUN apt-get update && apt-get install -y \
+RUN apt-get update -q && apt-get install -q -y --no-install-recommends \
+  gcc \
```
Comment (ejoerns) [inline test/Dockerfile:5]: These need to be added since it was installed as a recommended package before?
Maybe give a short hint in the commit message.

File: test/Dockerfile [hunk 3]
```diff
@@ -12,21 +14,31 @@ RUN apt-get update && apt-get install -y \
   libfdisk-dev \
   libnl-genl-3-dev
 
+# Required for building the cgi example
```
Comment (ejoerns) [inline test/Dockerfile:17]: Shouldn't this be a separate change?

File: test/Dockerfile [hunk 4]
```diff
@@ -75,6 +76,7 @@ RUN mkdir -p /lib/modules /var/run/dbus
 # Remove apt lists
 RUN rm -rf /var/lib/apt/lists/*
 
-RUN adduser --disabled-password --gecos "" rauc-ci
+RUN adduser --disabled-password --gecos "" rauc-ci && \
+    echo "rauc-ci ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/ptx
```
Comment (ejoerns) [inline test/Dockerfile:83]: Maybe use a different file name here?

File: .github/workflows/container.yml [hunk 5]
```diff
@@ -6,13 +6,18 @@ on:
...
     paths:
+      - '.github/workflows/container.yml'
       - 'test/Dockerfile'
   workflow_dispatch: {}
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
+    - name: Install QEMU
```
Comment (ejoerns) [inline .github/workflows/container.yml:17]: This seems to be unrelated to `build on workflow changes`, thus maybe put it in another commit.

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@ jobs:
   build:
     runs-on: ubuntu-latest
     container:
-      image: ghcr.io/rauc/rauc/rauc-ci:latest
+      image: ghcr.io/jluebbe/rauc/rauc-ci:latest
```
Comment (ejoerns) [inline .github/workflows/tests.yml:9]: This doesn't match the commit title (which still includes `WIP` anyway).

**Review by ejoerns — CHANGES_REQUESTED** [review]
Looks like a great enhancement for CI testing :+1:

Two minor notes on commit messages:

* `test/Dockerfile: only build desync on x86_64` should be `test/Dockerfile: build desync only on x86_64`

* `.github/workflows/container: add multiimage debugging` -> `multi-image` is better to read

File: test/Dockerfile (hunk 1 above)
Comment (jluebbe) [reply test/Dockerfile]: `--comment` is only supported from bookworm. At that point of the series, it's still bullseye.

File: test/Dockerfile (hunk 3 above)
Comment (jluebbe) [reply test/Dockerfile:17]: No, as this is only necessary due to moving to explicitly installing dependencies (`--no-install-recommends`).

File: .github/workflows/tests.yml (hunk 6 above)
Comment (jluebbe) [reply .github/workflows/tests.yml:9]: l'll fix the commit message.

File: .github/workflows/container.yml (hunk 5 above)
Comment (jluebbe) [reply .github/workflows/container.yml:17]: Move to `.github/workflows/container: build multi-platform image`.

File: .github/workflows/tests.yml (hunk 6 above)
Comment (jluebbe) [reply .github/workflows/tests.yml:9]: Yes, see the last paragraph in the description. :)

Comment (ejoerns) [reply .github/workflows/tests.yml:9]: Got it, thx!

**Review by ejoerns — DISMISSED** [review]

**Comment by jluebbe** [comment]
I've removed the WIP commit.

**Review by ejoerns — APPROVED** [review]
